### PR TITLE
add timeout support for handlers

### PR DESCRIPTION
**Pull request by lobeck**

### CODE REVIEW

File: lib/puppet/provider/sensu_handler/json.rb [hunk 1]
```diff
@@ -135,4 +135,12 @@ def subdue
...
   end
+
+  def timeout
+    conf['handlers'][resource[:name]]['timeout']
+  end
+
+  def timeout=(value)
+    conf['handlers'][resource[:name]]['timeout'] = value
+  end
 end
```
Comment (jaxxstorm) [inline lib/puppet/provider/sensu_handler/json.rb]: Can you add a trailing newline here?

File: manifests/handler.pp [hunk 2]
```diff
@@ -62,6 +62,10 @@
 #   Hash.  Handler subdue configuration
 #   Default: undef
 #
+# [*timeout*]
+#   Hash.  Handler timeout configuration
+#   Default: 10
```
Comment (jaxxstorm) [inline manifests/handler.pp]: default is undef as far as I can see?

Comment (lobeck) [inline manifests/handler.pp]: You're right, needs to be undef. We've taken the value from the sensu documentation, but the default only applies for tcp handlers (https://github.com/sensu/sensu/blob/master/lib/sensu/server/handle.rb#L63)

Will switch to `undef` here.

**Comment by jaxxstorm** [comment]
@lobeck can you rebase this and fix the conflicts?

**Comment by lobeck** [comment]
@jaxxstorm done.

**Pull request merged**